testing/simapp/app.go

```diff
@@ -465,7 +466,31 @@ func NewSimApp(
 		),
 	)
 
-	app.WasmClientKeeper = wasmkeeper.NewKeeper(appCodec, keys[wasmtypes.StoreKey], authtypes.NewModuleAddress(govtypes.ModuleName).String())
+	// 08-wasm's Keeper can be instantiated in two different ways:
+	// 1. If the chain uses x/wasm:
+	// Both x/wasm's Keeper and 08-wasm Keeper should share the same Wasm VM instante.
+	// - Instantiate the Wasm VM in app.go with the parameters of your choice.
+	// - Create an Option with this Wasm VM instance (see https: //github.com/CosmWasm/wasmd/blob/db93d7b6c7bb6f4a340d74b96a02cec885729b59/x/wasm/keeper/options.go#L21-L25).
+	// - Pass the option to the x/wasm NewKeeper contructor function (https://github.com/CosmWasm/wasmd/blob/db93d7b6c7bb6f4a340d74b96a02cec885729b59/x/wasm/keeper/keeper_cgo.go#L36).
+	// - Pass a pointer to the Wasm VM instance to 08-wasm NewKeeper constructor function.
+	//
+	// 2. If the chain does not use x/wasm:
+	// Even though it is still possible to use method 1 above
+	// (e.g. instantiating a Wasm VM in app.go an pass it in 08-wasm NewKeeper),
+	// since there is no need to share the Wasm VM instance with another module
+	// you can use NewKeeperWithConfig constructor function and provide
+	// the Wasm VM configuration parameters of your choice.
+	// Check out the WasmConfig type definition for more information on
+	// each parameter. Some parameters allow node-leve configurations.
+	//
+	// In the code below we use the second method because we are not using x/wasm in this app.go.
+	wasmConfig := wasmtypes.WasmConfig{
+		DataDir:           "ibc_08-wasm_client_data",
+		SupportedFeatures: "iterator",
```

Comment:
haven't had time to look into it atm but leaving as a Q. How important are these features? What if a chain that uses `x/wasm` has a keeper set up w/o this supported feature?

Reply:
It's a good question, @DimitrisJim. I also haven't looked extensively into this, but I have seen that the VM in `x/wasm` is configured using all available capabilities (see [here](https://github.com/CosmWasm/wasmd/blob/5d05b85875ee4f6befc69c843f6915df5bd68ab9/app/app.go#L586) and [here](https://github.com/CosmWasm/wasmd/blob/main/app/wasm.go#L6)). Looks like 08-wasm only needs the `iterator` (at least that's the one that was used in the code we got from Strangelove; there was another one -`storage`-, that apparently does not exist, so I removed it).